### PR TITLE
CDRIVER-4056 Send LB in handshake and reject non-LB connections

### PR DESCRIPTION
**Pull request by kevinAlbs**
- Add loadBalanced to the handshake.
- Reject server connections that do not include loadBalanced in the reply.

### CODE REVIEW

File: src/libmongoc/src/mongoc/mongoc-server-description-private.h [hunk 1]
```diff
@@ -183,4 +184,11 @@ void
 mongoc_server_description_set_topology_version (mongoc_server_description_t *sd,
                                                 const bson_t *tv);
 
+/* If a service_id is set on the topology description, copies it to @oid and
+ * returns true. Otherwise returns false and zeros out oid.
+ */
+bool
+mongoc_server_description_service_id (
+   const mongoc_server_description_t *description, bson_oid_t *oid);
```
Comment (chardan) [inline src/libmongoc/src/mongoc/mongoc-server-description-private.h:192]: nit: It's probably overly-pedantic, as basically nobody writes this, but did you mean:

bool
mongoc_server_description_service_id (
   const mongoc_server_description_t * const description, bson_oid_t *oid);

**Review by chardan — APPROVED** [review]
This looks good to me! I'm going a little bit on faith that the tests test what they're supposed to test (looks like  they do)...

File: src/libmongoc/doc/errors.rst [hunk 2]
```diff
@@ -31,6 +31,8 @@ Many C Driver functions report errors by returning ``false`` or -1 and filling o
...
 +-----------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+|                                         | ``MONGOC_ERROR_CLIENT_INVALID_LOAD_BALANCER``                                                                                   | You attempted to connect to a MongoDB server behind a load balancer, but the server does not advertise load balanced support.                                                                                                                                                                                                              |
```
Comment (benjirewis) [inline src/libmongoc/doc/errors.rst:34]: British spelling for `advertise` but I assume this error message was copied from somewhere.

**Review by benjirewis — APPROVED** [review]
LGTM, just a nit.

File: src/libmongoc/doc/errors.rst (hunk 2 above)
Comment (kevinAlbs) [reply src/libmongoc/doc/errors.rst:34]: TIL I've been using the British spelling. Updated.

File: src/libmongoc/src/mongoc/mongoc-server-description-private.h (hunk 1 above)
Comment (kevinAlbs) [reply src/libmongoc/src/mongoc/mongoc-server-description-private.h:192]: I did not. I do not think a const pointer adds a guarantee from the caller's perspective.

If `mongoc_server_description_service_id` promises not to change what description points to, a caller could not observe it.

Comment (chardan) [reply src/libmongoc/src/mongoc/mongoc-server-description-private.h:192]: +1. And I don't think it really buys anything, either.

**Review by chardan — COMMENTED** [review]
+1 LGTM

**Pull request merged**